frontend_tests/node_tests/upload.js

```diff
@@ -20,177 +20,489 @@ global.document.location.host = 'foo.com';
+                    assert.equal(Object.keys(params.locale.strings).length, 1);
+                    assert("timedOut" in params.locale.strings);
+                } else if (func_name === "ProgressBar") {
+                    uppy_used_progressbar = true;
+                    assert.equal(params.target, '#compose-send-status');
+                    assert.equal(params.hideAfterFinish, false);
+                } else {
+                    /* istanbul ignore next */
+                    assert.fail(`Missing tests for ${func_name}`);
+                }
+            },
+            on: () => {},
         };
     }
+    uppy_stub.Plugin = plugin_stub;
+    zrequire('upload', undefined, {'@uppy/core': uppy_stub});
+    upload.setup_upload({mode: "compose"});
 
-    function assert_side_effects(msg) {
-        assert($("#compose-send-status").hasClass("alert-error"));
-        assert(!$("#compose-send-status").hasClass("alert-info"));
-        assert.equal($("#compose-send-button").prop("disabled"), false);
-        assert.equal($("#compose-error-msg").text(), msg);
-    }
+    assert.equal(uppy_stub_called, true);
+    assert.equal(uppy_set_meta_called, true);
+    assert.equal(uppy_used_xhrupload, true);
+    assert.equal(uppy_used_progressbar, true);
 
-    function test(err, msg, server_response = null, file = {}) {
-        setup_test();
-        file.trackingId = "1549958107000";
-        upload_opts.error(err, server_response, file);
-        assert_side_effects(msg);
-    }
+});
 
-    const msg_prefix = 'translated: ';
-    const msg_1 = 'File upload is not yet available for your browser.';
-    const msg_2 = 'Unable to upload that many files at once.';
-    const msg_3 = '"foobar.txt" was too large; the maximum file size is 25MB.';
-    const msg_4 = 'Sorry, the file was too large.';
-    const msg_5 = 'An unknown error occurred.';
-    const msg_6 = 'File and image uploads have been disabled for this organization.';
-
-    test('BrowserNotSupported', msg_prefix + msg_1);
-    test('TooManyFiles', msg_prefix + msg_2);
-    test('FileTooLarge', msg_prefix + msg_3, null, {name: 'foobar.txt'});
-    test(413, msg_prefix + msg_4);
-    test(400, 'ちょっと…', {msg: 'ちょっと…'});
-    test('Do-not-match-any-case', msg_prefix + msg_5);
-
-    // If uploading files has been disabled, then a different error message is
-    // displayed when a user tries to paste or drag a file onto the UI.
-    page_params.max_file_upload_size = 0;
-    test('FileTooLarge', msg_prefix + msg_6, null);
+run_test('file_input', () => {
+    set_global('$', global.make_zjquery());
+    zrequire('upload', undefined, {'@uppy/core': global_uppy_stub});
+
+    upload.setup_upload({mode: "compose"});
+
+    const change_handler = $("body").get_on_handler("change", "#file_input");
+    const files = ["file1", "file2"];
+    const event = {
+        target: {
+            files: files,
+        },
+    };
+    var upload_files_called = false;
```

Comment:
You’ve undone my `var` → `let` updates so this doesn’t pass `lint` anymore, but I’ll assume you’re on it.